src/videojs-hls.js

```diff
@@ -736,8 +736,14 @@ videojs.Hls.prototype.drainBuffer = function(event) {
   this.segmentParser_.flushTags();
 
   tags = [];
+  segment.preciseTimestamp = -1;
```

Comment:
why not default it `null`? Then below on line `890` you only need to check for truthyness.